parser/diff.go

```diff
@@ -0,0 +1,130 @@
+package parser
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/reviewdog/reviewdog/diff"
+	"github.com/reviewdog/reviewdog/filter"
+	"github.com/reviewdog/reviewdog/proto/rdf"
+)
+
+var _ Parser = &DiffParser{}
+
+type DiffParser struct {
+	strip int
+	wd    string
+}
+
+func NewDiffParser(strip int) *DiffParser {
+	p := &DiffParser{strip: strip}
+	p.wd, _ = os.Getwd()
+	return p
+}
+
+func (p *DiffParser) Parse(r io.Reader) ([]*rdf.Diagnostic, error) {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
exported method DiffParser.Parse should have comment or be unexported